examples/energy-management-app/energy-management-common/include/DeviceEnergyManagementDelegateImpl.h

```diff
@@ -18,69 +18,276 @@
 
 #pragma once
 
-#include "app/clusters/device-energy-management-server/device-energy-management-server.h"
-
+#include <app/clusters/device-energy-management-server/device-energy-management-server.h>
 #include <app/util/config.h>
-#include <cstring>
 
 using chip::Protocols::InteractionModel::Status;
 namespace chip {
 namespace app {
 namespace Clusters {
 namespace DeviceEnergyManagement {
 
+class DEMManufacturerDelegate;
+
 /**
  * The application delegate.
  */
 class DeviceEnergyManagementDelegate : public DeviceEnergyManagement::Delegate
 {
 public:
+    DeviceEnergyManagementDelegate();
+
+    void SetDeviceEnergyManagementInstance(DeviceEnergyManagement::Instance & instance);
+
+    void SetDEMManufacturerDelegate(DEMManufacturerDelegate & deviceEnergyManagementManufacturerDelegate);
+
+    /**
+     *
+     * Implement the DeviceEnergyManagement::Delegate interface
+     *
+     */
+
+    /**
+     * @brief Implements a handler to begin to adjust client power
+     *        consumption/generation to the level requested.
+     *
+     * @param power Milli-Watts the ESA SHALL use during the adjustment period.
+     * @param duration The duration that the ESA SHALL maintain the requested power for.
+     * @return  Success if the adjustment is accepted; otherwise the command SHALL be rejected with appropriate error.
+     */
     virtual Status PowerAdjustRequest(const int64_t power, const uint32_t duration, AdjustmentCauseEnum cause) override;
```

Comment:
set unit of measurement to arguments. if power is in milliwatts, it should be power_mW or something like that. Duration also needs a measure unit (the documentation does not even say if this is hours, minutes, seconds or milliseconds).

Applies throughout: all unit-integers should have unit attached to the names.